src/main/java/org/gridsuite/modification/server/modifications/OperatingStatusModification.java

```diff
@@ -0,0 +1,176 @@
+        if (connectAllTerminals(equipment)) {
+            equipment.newExtension(OperatingStatusAdder.class).withStatus(OperatingStatus.Status.IN_OPERATION).add();
+        } else {
+            throw new NetworkModificationException(OPERATING_ACTION_ERROR, "Unable to connect all equipment ends");
+        }
+
+        subReporter.report(Report.builder()
+                .withKey("switchOn" + equipmentTypeName + "Applied")
+                .withDefaultMessage(equipmentTypeName + " ${id} (id) : switch on applied")
+                .withValue("id", equipment.getId())
+                .withSeverity(TypedValue.INFO_SEVERITY)
+                .build());
+    }
+
+    private void applyEnergiseEquipmentEnd(Reporter subReporter, Identifiable<?> equipment, String equipmentTypeName, TwoSides side) {
+        if (equipment instanceof Branch<?> branch) {
+            TwoSides oppositeSide = side == TwoSides.ONE ? TwoSides.TWO : TwoSides.ONE;
+            if (connectOneTerminal(branch.getTerminal(side)) && disconnectOneTerminal(branch.getTerminal(oppositeSide))) {
+                branch.newExtension(OperatingStatusAdder.class).withStatus(OperatingStatus.Status.IN_OPERATION).add();
+            } else {
+                throw new NetworkModificationException(OPERATING_ACTION_ERROR, "Unable to energise equipment end");
+            }
+
+            subReporter.report(Report.builder()
+                    .withKey("energise" + equipmentTypeName + "EndApplied")
+                    .withDefaultMessage(equipmentTypeName + " ${id} (id) : energise the side ${side} applied")
+                    .withValue("id", equipment.getId())
+                    .withValue("side", side.name())
+                    .withSeverity(TypedValue.INFO_SEVERITY)
+                    .build());
+        }
+    }
+
+    private boolean disconnectAllTerminals(Identifiable<?> equipment) {
+        return ModificationUtils.getInstance().getTerminalsFromIdentifiable(equipment).stream().allMatch(this::disconnectOneTerminal);
+    }
+
+    private boolean disconnectOneTerminal(Terminal terminal) {
+        return !terminal.isConnected() || terminal.disconnect();
+    }
+
+    private boolean connectAllTerminals(Identifiable<?> equipment) {
+        return ModificationUtils.getInstance().getTerminalsFromIdentifiable(equipment).stream().allMatch(this::connectOneTerminal);
+    }
+
+    private boolean connectOneTerminal(Terminal terminal) {
+        return terminal.isConnected() || terminal.connect();
+    }
+
+    public Tripping getTrippingFromIdentifiable(Identifiable<?> identifiable) {
+        if (identifiable instanceof Branch<?> branch) {
+            return new BranchTripping(branch.getId());
+        } else if (identifiable instanceof ThreeWindingsTransformer w3t) {
+            return new ThreeWindingsTransformerTripping(w3t.getId());
+        } else if (identifiable instanceof HvdcLine hvdcLine) {
```

Comment:
How about TieLine, DanglingLine?

Reply:
okey for the DanglingLine
TieLine is instanceof  Branch